crates/libs/windows/src/core/hstring.rs

```diff
@@ -402,11 +402,6 @@ struct Header {
     _0: u32,
     _1: u32,
     data: *mut u16,
-    shared: core::mem::MaybeUninit<Shared>,
-}
-
-#[repr(C)]
-struct Shared {
     count: RefCount,
     buffer_start: u16,
```

Comment:
I'm guessing this is this way to match the way things already were in COM-land, but it's a little weird to me that `buffer_start` is not zero-sized.

Reply:
Not having this as a zero-sized type caused some confusion that I was discussing with @kennykerr - namely, that we allocate `std::mem::size_of::<Header> + 2 * length_of_hstring` bytes we're allocating enough space for string data that fits all of the contents accounted for by `length_of_hstring` plus a terminating null `u16` which `length_of_string` does not account for but that extra `u16` in `Header` does. 

Definitely something to consider...